hypothesis-python/src/hypothesis/database.py

```diff
@@ -278,6 +278,9 @@ def __init__(self) -> None:
     def __repr__(self) -> str:
         return f"InMemoryExampleDatabase({self.data!r})"
 
+    def __eq__(self, other: object) -> bool:
+        return isinstance(other, InMemoryExampleDatabase) and self.data == other.data
```

Comment:
Ah, here I think we actually want `self.data is other.data`; the equality semantics are not "currently has the same contents" but rather "is reasonably expected to _always_ have the same contents".